consensus/dbft/recovery_message.go

```diff
@@ -43,153 +40,59 @@ type (
-	if hasReq {
-		m.prepareRequest = &message{}
-		m.prepareRequest.DecodeBinary(r)
-		if r.Err == nil && m.prepareRequest.Type != prepareRequestType {
-			r.Err = errors.New("recovery message PrepareRequest has wrong type")
-			return
-		}
-	} else {
-		l := r.ReadVarUint()
-		if l != 0 {
-			if l == util.Uint256Size {
-				m.preparationHash = new(util.Uint256)
-				r.ReadBytes(m.preparationHash[:])
-			} else {
-				r.Err = errors.New("invalid data")
-			}
-		} else {
-			m.preparationHash = nil
-		}
-	}
-
-	r.ReadArray(&m.preparationPayloads)
-	r.ReadArray(&m.commitPayloads)
-}
-
-// EncodeBinary implements the io.Serializable interface.
-func (m *recoveryMessage) EncodeBinary(w *io.BinWriter) {
-	w.WriteArray(m.changeViewPayloads)
-
-	hasReq := m.prepareRequest != nil
-	w.WriteBool(hasReq)
-	if hasReq {
-		m.prepareRequest.EncodeBinary(w)
-	} else {
-		if m.preparationHash == nil {
-			w.WriteVarUint(0)
-		} else {
-			w.WriteVarUint(util.Uint256Size)
-			w.WriteBytes(m.preparationHash[:])
-		}
-	}
-
-	w.WriteArray(m.preparationPayloads)
-	w.WriteArray(m.commitPayloads)
-}
-
-// DecodeBinary implements the io.Serializable interface.
-func (p *changeViewCompact) DecodeBinary(r *io.BinReader) {
-	p.ValidatorIndex = r.ReadB()
-	p.OriginalViewNumber = r.ReadB()
-	p.Timestamp = r.ReadU64LE()
-	p.InvocationScript = r.ReadVarBytes(1024)
-}
-
-// EncodeBinary implements the io.Serializable interface.
-func (p *changeViewCompact) EncodeBinary(w *io.BinWriter) {
-	w.WriteB(p.ValidatorIndex)
-	w.WriteB(p.OriginalViewNumber)
-	w.WriteU64LE(p.Timestamp)
-	w.WriteVarBytes(p.InvocationScript)
-}
-
-// DecodeBinary implements the io.Serializable interface.
-func (p *commitCompact) DecodeBinary(r *io.BinReader) {
-	p.ViewNumber = r.ReadB()
-	p.ValidatorIndex = r.ReadB()
-	r.ReadBytes(p.Signature[:])
-	p.InvocationScript = r.ReadVarBytes(1024)
-}
-
-// EncodeBinary implements the io.Serializable interface.
-func (p *commitCompact) EncodeBinary(w *io.BinWriter) {
-	w.WriteB(p.ViewNumber)
-	w.WriteB(p.ValidatorIndex)
-	w.WriteBytes(p.Signature[:])
-	w.WriteVarBytes(p.InvocationScript)
-}
-
-// DecodeBinary implements the io.Serializable interface.
-func (p *preparationCompact) DecodeBinary(r *io.BinReader) {
-	p.ValidatorIndex = r.ReadB()
-	p.InvocationScript = r.ReadVarBytes(1024)
-}
-
-// EncodeBinary implements the io.Serializable interface.
-func (p *preparationCompact) EncodeBinary(w *io.BinWriter) {
-	w.WriteB(p.ValidatorIndex)
-	w.WriteVarBytes(p.InvocationScript)
-}
-
 // AddPayload implements the payload.RecoveryMessage interface.
 func (m *recoveryMessage) AddPayload(p payload.ConsensusPayload) {
 	validator := uint8(p.ValidatorIndex())
 
 	switch p.Type() {
 	case payload.PrepareRequestType:
-		m.prepareRequest = &message{
+		pr := p.GetPrepareRequest().(*prepareRequest)
+		m.PrepareRequest = &message{
 			Type:       prepareRequestType,
 			ViewNumber: p.ViewNumber(),
-			payload:    p.GetPrepareRequest().(*prepareRequest),
+			msgPayload: pr,
```

Comment:
Less changes in code is better:
```suggestion
			msgPayload: p.GetPrepareRequest().(*prepareRequest),
```